### PR TITLE
Support for more TensorScatter* operations

### PR DESCRIPTION
**Pull request by javidcf**
Added support for TensorScatterMax, TensorScatterMin and TensorScatterSub.

Added tests for all TensorScatter* operations.

### CODE REVIEW

File: tests/test_backend.py [hunk 1]
```diff
@@ -6102,5 +6102,70 @@ def func(x):
...
+        tensor_val = make_xval([3, 4, 5])
+        indices_val = np.array([[2, 3], [0, 1]], np.int32)
+        updates_val = make_xval([2, 5]) + 3
+        self._run_test_case(func, [_OUTPUT], {_INPUT: tensor_val, _INPUT1: indices_val, _INPUT2: updates_val})
+
+    @check_tf_min_version("1.12.1")
+    @check_opset_min_version(11, "ScatterND")
+    @skip_tfjs("not supported in tfjs")
+    def test_tensor_scatter_update(self):
+        def func(tensor, indices, updates):
```
Comment (fatcat-z) [inline tests/test_backend.py:6161]: This has already been covered by existing test case, so not necessary.

File: tests/test_backend.py [hunk 2]
```diff
@@ -6102,5 +6102,70 @@ def func(x):
         x_val = make_xval([2, 3])
         self._run_test_case(func, [_OUTPUT], {_INPUT: x_val})
 
+    @check_tf_min_version("1.12.1")
+    @check_opset_min_version(16, "ScatterND")
+    @skip_tfjs("not supported in tfjs")
+    def test_tensor_scatter_add(self):
```
Comment (fatcat-z) [inline tests/test_backend.py:6108]: This has already been covered by existing test case, so not necessary.

File: tests/test_backend.py [hunk 3]
```diff
@@ -6102,5 +6102,70 @@ def func(x):
...
+
+        tensor_val = make_xval([3, 4, 5])
+        indices_val = np.array([[1, 2], [0, 1]], np.int32)
+        updates_val = make_xval([2, 5]) + 3
+        self._run_test_case(func, [_OUTPUT], {_INPUT: tensor_val, _INPUT1: indices_val, _INPUT2: updates_val})
+
+    @check_tf_min_version("2.3.0")
+    @check_opset_min_version(16, "ScatterND")
+    @skip_tfjs("not supported in tfjs")
+    def test_tensor_scatter_max(self):
```
Comment (fatcat-z) [inline tests/test_backend.py:6108]: Could you please add couple of test case to cover the scenario that indices dtype is int64?

File: tests/test_backend.py [hunk 4]
```diff
@@ -6102,5 +6102,70 @@ def func(x):
...
+    @check_tf_min_version("2.3.0")
+    @check_opset_min_version(16, "ScatterND")
+    @skip_tfjs("not supported in tfjs")
+    def test_tensor_scatter_min(self):
+        def func(tensor, indices, updates):
+            op = tf.tensor_scatter_nd_min(tensor, indices, updates)
+            return tf.identity(op, name=_TFOUTPUT)
+
+        tensor_val = make_xval([3, 4, 5])
+        indices_val = np.array([[2, 3], [0, 1]], np.int32)
```
Comment (fatcat-z) [inline tests/test_backend.py:6127]: Could you please add couple of test case to cover the scenario that indices dtype is int64?

File: tests/test_backend.py [hunk 5]
```diff
@@ -6102,5 +6102,70 @@ def func(x):
...
+    @check_tf_min_version("1.12.1")
+    @check_opset_min_version(16, "ScatterND")
+    @skip_tfjs("not supported in tfjs")
+    def test_tensor_scatter_sub(self):
+        def func(tensor, indices, updates):
+            op = tf.tensor_scatter_nd_sub(tensor, indices, updates)
+            return tf.identity(op, name=_TFOUTPUT)
+
+        tensor_val = make_xval([3, 4, 5])
+        indices_val = np.array([[2, 3], [0, 1]], np.int32)
```
Comment (fatcat-z) [inline tests/test_backend.py:6140]: Could you please add couple of test case to cover the scenario that indices dtype is int64?

**Comment by javidcf** [comment]
Seems there was some issue with Azure in the failing checks above. Can those be retried @fatcat-z ?

**Review by fatcat-z — APPROVED** [review]
LGTM, thanks for your contributions!

**Pull request merged**